### DISCUSSION TITLE
Get Playlist and PlaylistItems missing fields for Track

### DISCUSSION THREAD

**Discussion by ulpcan** [spotify-web-api-java/spotify-web-api-java#333] [Ideas]
Hello,

PlaylistTrack has IPlaylistItem track field;

Inside IPlaylistItem track interface we need additional information for externalIds, also there are more fields are avaliable in Spotify response. I don't know whether Episode has those fields but certainly Track has.

Screenshot for spotify response and interface:

https://ibb.co/0cyQDnW

Could you please take a look at it and add in the future?

Thanks for your time :+1: :100: :pray:

**Comment by dargmuesli** [comment]
Maybe it helps to type cast the playlist item to Track? We had a few other threads on this repo regarding that, you may find them easily.

**Reply by ulpcan** [reply]
Thanks a lot. :)

**Reply by AurJur** [reply]
I also stumbled accross this issue. Even if the proposed casting technically works, usually a casting is not a desirable approach. Is there a reason for such design?

**Reply by dargmuesli** [reply]
That's a question that could well be directed at the developers at Spotify. Their API gives data that could either be this or that, one only really knows once its there, so casting was implemented.
